halo2/src/prover.rs

```diff
@@ -65,7 +65,7 @@ impl Halo2Prover {
         witness: &[(&str, Vec<F>)],
     ) -> Vec<u8> {
         // TODO this is hacky
```

Comment:
This comment can now be removed

Reply:
Isn't it still hacky to do this `leading_zero` magic?

Reply:
that we can replace by log2, but I think the hack was pointing to reading length from column